modules/service/src/main/java/org/mule/runtime/module/service/internal/manager/MuleServiceManager.java

```diff
@@ -67,15 +70,35 @@ private void startServices() throws MuleException {
     }
   }
 
+  /**
+   * Stop all discovered services. In the case of 'HTTP' and 'Scheduler' service, if present, they should be stopped last and in
+   * that order.
+   *
+   * @throws MuleException
+   */
   @Override
   public void stop() throws MuleException {
+    LinkedList<Service> servicesWithSpecialStoppingOrder = new LinkedList<>();
```

Comment:
use `new ArrayList<>(2)` or simply two variables.

Reply:
Changed to Optionals. Solution seems cleaner.

Reply:
Back to null.